packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -141,29 +212,46 @@ export class QueuedRequestController extends BaseController<
    * requests, ensuring they are executed one after the other to prevent concurrency issues and maintain proper
    * execution flow.
    *
-   * @param requestNext - A function representing the request to be enqueued. It returns a promise that
+   * @param request - The JSON-RPC request to process.
+   * @param requestNext - A function representing the next steps for processing this request. It returns a promise that
    * resolves when the request is complete.
    * @returns A promise that resolves when the enqueued request and any subsequent asynchronous
    * operations are fully processed. This allows you to await the completion of the enqueued request before continuing
    * with additional actions. If there are multiple enqueued requests, this function ensures they are processed in
    * the order they were enqueued, guaranteeing sequential execution.
    */
-  async enqueueRequest(requestNext: (...arg: unknown[]) => Promise<unknown>) {
+  async enqueueRequest(
+    request: QueuedRequestMiddlewareJsonRpcRequest,
+    requestNext: (...arg: unknown[]) => Promise<unknown>,
```

Comment:
It looks like there's nothing in the current implementation or tests that prevents this change: 
```suggestion
    requestNext: () => Promise<unknown>,
```
Or could this be a case where we want to be as permissive as possible about the callbacks that can be assigned to `requestNext` (and there aren't any downstream type constraints that could get in the way of this)?
```suggestion
    requestNext: (...arg: never[]) => Promise<unknown>,
```

Reply:
This change fixes this error: https://github.com/MetaMask/core/pull/3968/files#r1503256245, but I'm guessing this is not how we want to type `requestNext`.

```suggestion
    requestNext: (...arg: unknown[]) => Promise<void>,
```